sdk/cosmos/azure_data_cosmos/src/lib.rs

```diff
@@ -7,12 +10,35 @@
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
 #![cfg_attr(docsrs, feature(doc_cfg_hide))]
 
-mod authorization_policy;
 pub mod clients;
+pub(crate) mod constants;
 mod options;
+mod partition_key;
+pub(crate) mod pipeline;
+mod query;
+pub(crate) mod utils;
 
-/// Model types sent to and received from the Cosmos DB API.
 pub mod models;
 
 pub use clients::{CosmosClient, CosmosClientMethods};
 pub use options::*;
+pub use partition_key::*;
+pub use query::*;
+use serde::Serialize;
+
+/// A zero-sized marker type that indicates a null value.
+///
+/// Used when specifying `null` for [`PartitionKey`] or [`Query`] parameter, when [`Option`] is not appropriate (for example, if you never intend to pass a non-null value).
+pub struct NullValue;
+
+impl Serialize for NullValue {
+    /// Serializes the [`NullValue`].
+    ///
+    /// This will always produce the value `null` (or equivalent in whatever format the type is being serialized to).
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_none()
+    }
+}
```

Comment:
Would this be useful in `typespec_client_core`? We could always move it later, though. Just putting it out there.

Reply:
We do - or did? - have a serialization helper that serialized explicit nulls. Still in `main` at least.

Reply:
I'm not seeing what you're referring to in `main`, but I'm definitely open to promoting this to `typespec_client_core` if you think it's useful there.